circuits/types/field/src/lsb.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright (C) 2019-2022 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// The snarkVM library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkVM library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
+
+use super::*;
+use snarkvm_utilities::{FromBytes, ToBytes};
+use std::iter;
+
+impl<E: Environment> LSB for Field<E> {
+    type Boolean = Boolean<E>;
+
+    /// Returns the LSB of the integer.
+    fn lsb(&self) -> Self::Boolean {
+        let mut bytes = match self.eject_value().to_bytes_le() {
```

Comment:
if possible we should use `to_bits` here, to avoid dealing with weird byte padding issues.